src/uct/ib/rc/accel/rc_mlx5.h

```diff
@@ -37,13 +37,28 @@ typedef struct uct_rc_mlx5_ep {
     uct_rc_mlx5_mp_context_t mp;
 } uct_rc_mlx5_ep_t;
 
+
+typedef enum {
+    UCT_RC_MLX5_EP_ADDR_TM_QP_NUM = UCS_BIT(0),
+    UCT_RC_MLX5_EP_ADDR_ECE       = UCS_BIT(1),
+    UCT_RC_MLX5_EP_ADDR_SR        = UCS_BIT(2),
+    UCT_RC_MLX5_EP_ADDR_TM_ECE    = UCS_BIT(3),
+    UCT_RC_MLX5_EP_ADDR_TM_SR     = UCS_BIT(4),
+} uct_rc_mlx5_ep_addr_flag_t;
+
+
 typedef struct uct_rc_mlx5_ep_address {
     uct_ib_uint24_t  qp_num;
-    /* For RNDV TM enabling 2 QPs should be created, one is for sending WRs and
-     * another one for HW (device will use it for RDMA reads and sending RNDV
-     * Complete messages). */
-    uct_ib_uint24_t  tm_qp_num;
     uint8_t          atomic_mr_id;
+    uint8_t          flags;    /* uct_rc_mlx5_ep_addr_flag_t */
+    /* Optional fields.
+     * For RNDV TM enabling 2 QPs should be created, one is for sending WRs and
+     * another one for HW (device will use it for RDMA reads and sending RNDV
+     * Complete messages).
+     * - uct_ib_uint24_t  tm_qp_num
+     * - uint32_t         tm_ece
+     * - uint32_t         ece
+     */
```

Comment:
need to avoid breaking wire compatibility
- pass address len from ucp
- add TODO to remove tm_qp when updating address version